api/src/user/dao/user.dao.ts

```diff
@@ -138,4 +138,73 @@ export class UserDao {
       where: { id: userId },
     });
   }
+
+  async getUserStats(userId: string) {
+    const [listCount, sharedListCount, movieCount, commentCount] =
+      await Promise.all([
+        this.prisma.user.findUniqueOrThrow({
+          where: { id: userId },
+          select: {
+            _count: {
+              select: {
+                list: {
+                  where: {
+                    AND: [{ creatorId: userId }],
+                  },
+                },
+              },
+            },
+          },
+        }),
+
+        this.prisma.user.findUniqueOrThrow({
+          where: { id: userId },
+          select: {
+            _count: {
+              select: {
+                list: {
+                  where: {
+                    NOT: { creatorId: userId },
+                    user: { some: { id: userId } },
```

Comment:
We already filter by user in the top `where`, do we need it again?

Reply:
Yes otherwise it doesn't return the proper results.